### PR TITLE
Protocol-spec transfer - Closes #4181

### LINKED ISSUE #4181: Protocol-spec transfer
### Description

- Basic Transfer acceptance
- Transfer transaction that is valid independently but as a block invalid (From account `B` -balance=0- transfer to account `C` before transferring from account `A` to `B`).
- Transfer transaction that is valid independently and as a block, but partially invalid ( Fund one account with 1 lisk and create two transfers with 0.5 lisk -not possible due to second transaction not having enough lisk after fee applied to first transaction)

### Acceptance Criteria

JSON generator for: 

- [x] Basic Transfer acceptance
- [x] Transfer transaction that is valid independently but as a block invalid.
- [x] Transfer transaction that is valid independently and as a block, but partially invalid

### Additional Information

Split from https://github.com/LiskHQ/lisk-sdk/issues/4087

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

Some new cases were needed for transfer transactions

### How did I solve it?

- By adding the missing cases
- Rewrote the generator using the ChainStateBuilder library

### How to manually test it?

- Check the output of the generators included in the PR

### Review checklist

- [x] The PR resolves #4181
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

File: protocol-specs/generators/block_processing_transfers/index.js [hunk 1]
```diff
@@ -2132,106 +90,142 @@ const accounts = {
...
+		passphrase:
+			'blame address tube insect cost knock major level regret bring april stick',
+		privateKey:
+			'b92e223981770c716ee54192a0ad028639d28d41221b72e455447bc4767aeb94caff2242b740a733daa3f3f96fc1592303b60c1704a8ac626e2704da039f41ee',
+		publicKey:
+			'caff2242b740a733daa3f3f96fc1592303b60c1704a8ac626e2704da039f41ee',
+		address: '2222471382442610527L',
+		balance: '0',
+	},
+	newAccount2: {
```
Comment (michielmulders) [inline protocol-specs/generators/block_processing_transfers/index.js]: Not sure if `newAccount2` is the best naming here? 😄

File: protocol-specs/generators/block_processing_transfers/index.js [hunk 2]
```diff
@@ -2132,106 +90,142 @@ const accounts = {
...
+const generateTestCasesValidBlockTransferTx = () => {
+	const chainStateBuilder = new ChainStateBuilder(
 		genesisBlock,
-		1,
-		0,
-		{
-			version: 1,
-			transactions: [],
-		},
+		initialAccountsState,
```
Comment (michielmulders) [inline protocol-specs/generators/block_processing_transfers/index.js]: Better readable here now, nice! 👍

File: protocol-specs/generators/block_processing_transfers/index.js [hunk 3]
```diff
@@ -2132,106 +90,142 @@ const accounts = {
...
+	const chainAndAccountStates = chainStateBuilder.getScenario();
+
 	return {
 		initialState: {
-			chain: [],
-			accounts: initialAccountState,
-		},
-		input: {
-			block,
+			chain: chainAndAccountStates.chain.slice(0, 1),
```
Comment (michielmulders) [inline protocol-specs/generators/block_processing_transfers/index.js]: Would be nice to add a comment why slice here?

Comment (pablitovicente) [reply protocol-specs/generators/block_processing_transfers/index.js]: Added comments for this case and the other one using `-1`

File: protocol-specs/generators/block_processing_transfers/index.js (hunk 1 above)
Comment (pablitovicente) [reply protocol-specs/generators/block_processing_transfers/index.js]: Changed names

File: protocol-specs/generators/block_processing_transfers/README.md [hunk 4]
```diff
@@ -1,13 +1,56 @@
...
 
-A set of test generators related to block processing.
-
-This set needs to be extended and improved.
-
-For now the following suites are generated:
-
-- Processing a new valid empty block
-- Processing a new valid block with one valid transfer transaction
+A set of fixture generatos for Transfer transactions
```
Comment (limiaspasdaniel) [inline protocol-specs/generators/block_processing_transfers/README.md]:
```suggestion
A set of fixture generators for Transfer transactions
```

File: protocol-specs/generators/block_processing_transfers/README.md [hunk 5]
```diff
@@ -1,13 +1,56 @@
...
+
+These generators makes use of an experimental library for combining transactions, include them into blocks and generate chain/account states.
```
Comment (limiaspasdaniel) [inline protocol-specs/generators/block_processing_transfers/README.md]:
```suggestion
These generators make use of an experimental library for combining transactions, include them into blocks and generate chain/account states.
```

**Pull request merged**